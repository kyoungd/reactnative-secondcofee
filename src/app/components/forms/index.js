import FormHeading from './FormHeading'
import SignInForm from './SignInForm'
import SignUpForm from './SignUpForm'
import ResetPasswordForm from './ResetPasswordForm'
import PersonalInformationForm from './PersonalInformationForm'
import ProfessionalExperienceForm from './ProfessionalExperienceForm'

export {
  FormHeading,
  SignInForm,
  SignUpForm,
  ResetPasswordForm,
  PersonalInformationForm,
  ProfessionalExperienceForm
}
