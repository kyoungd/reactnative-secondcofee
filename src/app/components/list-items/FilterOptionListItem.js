import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'

const Wrapper = styled.TouchableOpacity`
  justify-content: center;
  align-items: center;
  padding: 8px 16px;
  margin: 0px 12px 12px 0px;
  border-radius: 8;
  border-width: 1;
  border-color: ${props => (props.active === true ? '#000' : '#dee3ea')};
  background-color: ${props => (props.active === true ? '#000' : '#fff')};
`

const Label = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: ${props => (props.active === true ? '#fff' : '#000')};
  text-transform: uppercase;
`

export default function FilterOptionListItem({ active, label, onPress }) {
  return (
    <Wrapper active={active} onPress={() => onPress()}>
      <Label active={active}>{label}</Label>
    </Wrapper>
  )
}

FilterOptionListItem.propTypes = {
  active: PropTypes.bool.isRequired,
  label: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
}
