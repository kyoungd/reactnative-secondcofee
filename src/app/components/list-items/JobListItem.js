import React from 'react'
import PropTypes from 'prop-types'
import { Ionicons, Feather } from '@expo/vector-icons'
import styled from 'styled-components/native'

export const Wrapper = styled.View`
  flex-direction: row;
  padding: 8px;
  border-radius: 8;
  margin: 16px 16px 8px;
  background-color: #fff;
  border-width: 1;
  border-color: #dee3ea;
`

const Left = styled.TouchableOpacity`
  flex: 1;
  margin-right: 8;
`

const IconWrapper = styled.View`
  width: 30;
  align-items: center;
`

export const Title = styled.Text.attrs({
  numberOfLines: 1
})`
  font-family: 'SourceSansProRegular';
  font-size: 17;
  color: #000;
  text-transform: uppercase;
`

export const CompanyName = styled.Text.attrs({
  numberOfLines: 1
})`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: #346df1;
  margin: 2px 0px 4px;
`

const Bottom = styled.View`
  height: 40;
  flex-direction: row;
`

const BottomItemWrapper = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: center;
`

const BottomItemIcon = styled(Ionicons).attrs({
  size: 18,
  color: '#a5a5a5'
})``

export const BottomItemText = styled.Text.attrs({
  numberOfLines: 1
})`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: #a9a9a9;
  margin: 0px 4px 4px;
`

export default class JobListItem extends React.Component {
  state = {
    isSaved: false
  }

  toggleSaved() {
    const { isSaved: savedStatus } = this.state
    this.setState({
      isSaved: !savedStatus
    })
  }

  render() {
    const { title, companyName, postedAt, experience, onPress } = this.props
    const { isSaved } = this.state

    return (
      <Wrapper>
        <Left onPress={() => onPress()}>
          <Title>{title}</Title>
          <CompanyName>{companyName}</CompanyName>

          <Bottom>
            <BottomItemWrapper>
              <BottomItemIcon name="ios-time" />
              <BottomItemText>{postedAt}</BottomItemText>
            </BottomItemWrapper>

            <BottomItemWrapper>
              <BottomItemIcon name="ios-flash" />
              <BottomItemText>Experience: {experience}</BottomItemText>
            </BottomItemWrapper>
          </Bottom>
        </Left>

        <IconWrapper>
          {!isSaved && (
            <Feather
              name="bookmark"
              size={26}
              color="#a5a5a5"
              onPress={() => this.toggleSaved()}
            />
          )}
          {isSaved && (
            <Ionicons
              name="md-bookmark"
              size={30}
              color="#5bbff7"
              onPress={() => this.toggleSaved()}
            />
          )}
        </IconWrapper>
      </Wrapper>
    )
  }
}

JobListItem.propTypes = {
  title: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  postedAt: PropTypes.string.isRequired,
  experience: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
}
