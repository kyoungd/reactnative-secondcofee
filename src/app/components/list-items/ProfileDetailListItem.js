import React from 'react'
import PropTypes from 'prop-types'
import { Ionicons } from '@expo/vector-icons'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  flex-direction: row;
  padding: 16px;
`

const IconWrapper = styled.View`
  justify-content: center;
`

const Right = styled.View`
  flex: 1;
  margin-left: 16;
`

const Title = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 13;
  color: #346df1;
`

const Label = styled.Text.attrs({
  numberOfLines: 2
})`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: #000;
  margin-top: 4;
`

export default function ProfileDetailListItem({ iconName, title, content }) {
  return (
    <Wrapper>
      <IconWrapper>
        <Ionicons size={25} color="#a5a5a5" name={iconName} />
      </IconWrapper>

      <Right>
        <Title>{title}</Title>
        <Label>{content}</Label>
      </Right>
    </Wrapper>
  )
}

ProfileDetailListItem.propTypes = {
  iconName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired
}
