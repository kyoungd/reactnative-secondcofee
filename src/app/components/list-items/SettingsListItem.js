import React from 'react'
import PropTypes from 'prop-types'
import { Switch } from 'native-base'
import { Ionicons } from '@expo/vector-icons'
import styled from 'styled-components/native'

const Wrapper = styled.TouchableOpacity`
  flex-direction: row;
  height: 60;
  padding: 16px;
`

const LabelWrapper = styled.View`
  flex: 1;
  justify-content: center;
  margin: 0px 8px;
`

const Label = styled.Text.attrs({
  numberOfLines: 1
})`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: #242635;
  margin: 0px 0px 4px;
`

const IconWrapper = styled.View`
  width: 30;
  align-items: ${props =>
    props.positionValue === 'left' ? 'flex-start' : 'flex-end'};
`

export default function SettingsListItem({
  iconName,
  text,
  withSwitch,
  isActive,
  onPress,
  onSwitchValueChange
}) {
  return (
    <Wrapper onPress={() => !withSwitch && onPress()}>
      <IconWrapper positionValue="left">
        <Ionicons name={iconName} color="#346df1" size={25} />
      </IconWrapper>

      <LabelWrapper>
        <Label>{text}</Label>
      </LabelWrapper>

      {!withSwitch && (
        <IconWrapper>
          <Ionicons name="ios-arrow-forward" color="#000" size={20} />
        </IconWrapper>
      )}
      {withSwitch && (
        <Switch
          value={isActive}
          onValueChange={value => onSwitchValueChange(value)}
        />
      )}
    </Wrapper>
  )
}

SettingsListItem.propTypes = {
  iconName: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  withSwitch: PropTypes.bool,
  isActive: PropTypes.bool,
  onPress: PropTypes.func,
  onSwitchValueChange: PropTypes.func
}
SettingsListItem.defaultProps = {
  withSwitch: false,
  isActive: false,
  onPress: () => null,
  onSwitchValueChange: () => null
}
