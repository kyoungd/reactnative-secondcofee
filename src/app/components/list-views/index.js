import FilterOptionsListView from './FilterOptionsListView'
import JobsListView from './JobsListView'
import SavedJobsListView from './SavedJobsListView'

export { FilterOptionsListView, JobsListView, SavedJobsListView }
