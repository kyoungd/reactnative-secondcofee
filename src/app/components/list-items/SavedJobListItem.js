import React from 'react'
import PropTypes from 'prop-types'
import Swipeout from 'react-native-swipeout'
import styled from 'styled-components/native'
import { ImageProgressComponent, ActionButton } from '../ui'
import { Title, CompanyName } from './JobListItem'

const Wrapper = styled.TouchableOpacity`
  flex-direction: row;
  padding: 16px;
`

const Right = styled.View`
  flex: 1;
  height: 80;
  margin-left: 16;
`

const Bottom = styled.View`
  flex: 1;
  justify-content: flex-end;
`

const AppliedText = styled(CompanyName)`
  color: #278f37;
`

export default class SavedJobListItem extends React.Component {
  state = {
    isApplied: false
  }

  render() {
    const { photoURL, title, companyName, onPress, onRemovePress } = this.props
    const { isApplied } = this.state

    return (
      <Swipeout
        autoClose
        right={[
          {
            text: 'Remove',
            type: 'delete',
            onPress: () => onRemovePress()
          }
        ]}
        backgroundColor="transparent"
      >
        <Wrapper onPress={() => onPress()}>
          <ImageProgressComponent
            photoURL={photoURL}
            resizeMode="cover"
            style={{ width: 50, height: 50 }}
            imageStyle={{ borderRadius: 8 }}
          />

          <Right>
            <Title>{title}</Title>
            <CompanyName>{companyName}</CompanyName>

            <Bottom>
              {!isApplied && (
                <ActionButton
                  text="Apply"
                  onPress={() => this.setState({ isApplied: true })}
                />
              )}
              {isApplied && <AppliedText>Applied</AppliedText>}
            </Bottom>
          </Right>
        </Wrapper>
      </Swipeout>
    )
  }
}

SavedJobListItem.propTypes = {
  photoURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  onRemovePress: PropTypes.func.isRequired
}
