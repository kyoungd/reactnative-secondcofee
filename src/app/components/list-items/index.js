import FilterOptionListItem from './FilterOptionListItem'
import JobListItem from './JobListItem'
import ProfileDetailListItem from './ProfileDetailListItem'
import SavedJobListItem from './SavedJobListItem'
import SettingsListItem from './SettingsListItem'

export {
  FilterOptionListItem,
  JobListItem,
  ProfileDetailListItem,
  SavedJobListItem,
  SettingsListItem
}
