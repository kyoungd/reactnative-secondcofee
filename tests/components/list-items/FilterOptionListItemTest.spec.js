import React from 'react'
import { shallow } from 'react-native-testing-library'
import { FilterOptionListItem } from 'components/list-items'

test('Component renders correctly', () => {
  const { output } = shallow(
    <FilterOptionListItem active label="Filter" onPress={() => null} />
  )
  expect(output).toMatchSnapshot()
})
