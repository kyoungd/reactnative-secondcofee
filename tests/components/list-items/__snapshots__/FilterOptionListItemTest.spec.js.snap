// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<ForwardRef
  active={true}
  onPress={[Function]}
>
  <ForwardRef
    active={true}
  >
    Filter
  </ForwardRef>
</ForwardRef>
`;
